src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMNavigator.java

```diff
@@ -260,8 +262,14 @@ public void visitSources(@NonNull final SCMSourceObserver observer) throws IOExc
                         serverUrl, getPrivateTokenAsPlainText(webHookCredentials), null, getProxyConfig(serverUrl));
                 webHookUrl = GitLabHookCreator.getHookUrl(server, true);
             }
+            // we may check with web hooks but we don't check if the projects returned are null
+
             for (Project p : projects) {
                 count++;
+                if (p == null) {
+                    System.out.println("null project");
+                    continue;
+                }
```

Comment:
Loose the println and simply filter them.

```suggestion
            projects = projects.stream().filter(Objects::nonNull).collect(Collectors.toList());
            for (Project p : projects) {
                count++;
```